### ISSUE TITLE
[Docs]: Improve documentation for metadata

### ISSUE THREAD

**Issue by vitalets** [microsoft/playwright#34094]
### Page(s)

https://playwright.dev/docs/next/api/class-testconfig#test-config-metadata

#### 1. Current `metadata` example does not pass TypeScript check:
```ts
import { defineConfig } from '@playwright/test';

export default defineConfig({
  metadata: 'acceptance tests',
});
```
Result:
```
Type 'string' is not assignable to type 'Metadata'.
``` 

#### 2. Documentation should mention the [certain fields](https://github.com/microsoft/playwright/blob/9c14cccc24db0a486e5fab3f3f4f4be887eb2d39/packages/html-reporter/src/metadataView.tsx#L25) that are [allowed](https://github.com/microsoft/playwright/blob/9c14cccc24db0a486e5fab3f3f4f4be887eb2d39/packages/html-reporter/src/metadataView.tsx#L65) in `metadata`, e.g.
```ts
export type Metainfo = {
  'revision.id'?: string;
  'revision.author'?: string;
  'revision.email'?: string;
  'revision.subject'?: string;
  'revision.timestamp'?: number | Date;
  'revision.link'?: string;
  'ci.link'?: string;
  'timestamp'?: number
};
```

**Comment by dgozman** [comment]
We discussed this issue, and are not sure how to improve it.

**Comment by dgozman** [comment]
@vitalets We are open to bringing the commit info into the metadata under an option in the config. If that goes public, then we'll definitely document the fields and the html reporter will show them nicely. Would you like to contribute?

**Comment by vitalets** [comment]
> [@vitalets](https://github.com/vitalets) We are open to bringing the commit info into the metadata under an option in the config. If that goes public, then we'll definitely document the fields and the html reporter will show them nicely. Would you like to contribute?

Yes, I'm ready to contribute. Could you guide me, how to name / where to place the config option?

**Comment by dgozman** [comment]
> Yes, I'm ready to contribute. Could you guide me, how to name / where to place the config option?

Something like `TestConfig.populateGitInfo`, and we'll finalize/rename before the release after the API review.

**Comment by vitalets** [comment]
> Something like `TestConfig.populateGitInfo`, and we'll finalize/rename before the release after the API review.

Ok! When `TestConfig.populateGitInfo` is enabled, Playwright internally activates `GitCommitInfo` plugin, that populates metadata?

**Comment by dgozman** [comment]
> > Something like `TestConfig.populateGitInfo`, and we'll finalize/rename before the release after the API review.
> 
> Ok! When `TestConfig.populateGitInfo` is enabled, Playwright internally activates `GitCommitInfo` plugin, that populates metadata?

Exactly.

**Comment by vitalets** [comment]
I thought more about it and got some concerns. 
We are going to introduce one more test runner option (`TestConfig.populateGitInfo`) that actually populates another existing option (`metadata`), with kind of third-party data needed only for specific reporter (`html`). 
Playwright configuration is already rather complex and has many options. Maybe move this option to html reporter itself? Or to GitInfo plugin somehow?
Otherwise, how will the following code behave from the performance perspective - if I enable gitinfo, but don't use html reporter:
```ts
export default defineConfig({
  populateGitInfo: true,
  reporter: 'list'
});
```
@dgozman What do you think?

**Comment by dgozman** [comment]
We discussed this being an option of the html reporter, but that is not compatible with `merge-reports` workflow. There is no harm in populating git info without html - any custom reporter can use that data. In fact, json reporter already includes all of the metadata today. Since we do not have plugins in the public api, config option seems like the best solution.

I think we'd recommend to only enable it on CI anyway:
```ts
export default defineConfig({
  populateGitInfo: !!process.env.CI,
});
```
